flink-core/src/main/java/org/apache/flink/api/common/eventtime/NoOpWatermarkGenerator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.eventtime;
+
+import org.apache.flink.annotation.Public;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * An implementation of a {@link WatermarkGenerator} that periodically emits watermarks
+ * or generates only one final watermark at the end of input.
+ */
+@Public
+public final class NoOpWatermarkGenerator<E> implements WatermarkGenerator<E> {
```

Comment:
I would make it a private class in the `TimestampsAndWatermarksOperator`. Certainly it should not be a `@Public` class.

Moreover I thought we should rather have a class like:

```
/**
 * An implementation of a {@link WatermarkGenerator} that does not emit watermarks.
 */
public final class NoOpWatermarkGenerator<E> implements WatermarkGenerator<E> {
	@Override
	public void onEvent(E event, long eventTimestamp, WatermarkOutput output) {
	}

	@Override
	public void onPeriodicEmit(WatermarkOutput output) {
	}
}
```